### PR TITLE
Add amp-story embed modes

### PR DESCRIPTION
**Pull request by newmuis**
This allows us to have an "embed mode" that changes the functionality/appearance of a story.  It currently only adds a single mode that disables ads and most system UI.

/cc @gmajoulet but I might not have time to wait for his review, since he's on Paris time at the moment 😄

### CODE REVIEW

File: extensions/amp-story/0.1/amp-story.js [hunk 1]
```diff
@@ -1541,6 +1595,15 @@ export class AmpStory extends AMP.BaseElement {
     const pageToBeInserted = this.getPageById(pageToBeInsertedId);
     const pageToBeInsertedEl = pageToBeInserted.element;
 
+    const allowAutomaticAdInsertion = this.stateService_
+        .getState(StateType.ALLOW_AUTOMATIC_AD_INSERTION);
+
+    if (pageToBeInsertedEl.hasAttribute('ad') &&
```
Comment (calebcordry) [inline extensions/amp-story/0.1/amp-story.js]: We have a public method `isAd()` on the page now if you would prefer

File: extensions/amp-story/0.1/amp-story-auto-ads.js [hunk 2]
```diff
@@ -98,10 +102,19 @@ export class AmpStoryAutoAds extends AMP.BaseElement {
...
     /** @private {Object<string, string>} */
     this.config_ = {};
+
+    /** @private @const {!AmpStoryStateService} */
+    this.stateService_ = new AmpStoryStateService();
   }
 
   /** @override */
   buildCallback() {
+    const embedMode = this.parseEmbedMode_(this.win.location.hash);
```
Comment (calebcordry) [inline extensions/amp-story/0.1/amp-story-auto-ads.js]: this doesn't exist on `auto-ads`

File: extensions/amp-story/0.1/amp-story.js [hunk 3]
```diff
@@ -1541,6 +1595,15 @@ export class AmpStory extends AMP.BaseElement {
     const pageToBeInserted = this.getPageById(pageToBeInsertedId);
     const pageToBeInsertedEl = pageToBeInserted.element;
 
+    const allowAutomaticAdInsertion = this.stateService_
```
Comment (calebcordry) [inline extensions/amp-story/0.1/amp-story.js]: Is the idea with this section to be future proof? If we change the `auto-ads` component this shouldn't be necessary. I could see wanting it in the future if there becomes another way to insert ads, so Ill leave it up to you.

File: extensions/amp-story/0.1/amp-story.js (hunk 1 above)
Comment (newmuis) [reply extensions/amp-story/0.1/amp-story.js]: Done.

File: extensions/amp-story/0.1/amp-story.js (hunk 3 above)
Comment (newmuis) [reply extensions/amp-story/0.1/amp-story.js]: Yeah, for future-proofing.

**Pull request merged**